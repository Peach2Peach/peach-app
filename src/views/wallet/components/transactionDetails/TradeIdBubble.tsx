import { Bubble } from '../../../../components/bubble'
import { useNavigateToOfferOrContract } from '../../../../hooks/useNavigateToOfferOrContract'
import { contractIdToHex } from '../../../../utils/contract/contractIdToHex'

type Props = {
  trade: ContractSummary
}

export const TradeIdBubble = ({ trade }: Props) => {
  const goToContract = useNavigateToOfferOrContract(trade)
  const tradeId = contractIdToHex(trade.id)

  return (
    <Bubble color="primary-mild" iconId="info" onPress={goToContract}>
      {tradeId}
    </Bubble>
  )
}
