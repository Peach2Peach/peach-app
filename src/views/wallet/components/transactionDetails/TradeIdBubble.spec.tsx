import { fireEvent, render } from "test-utils";
import { contractSummary } from "../../../../../tests/unit/data/contractSummaryData";
import { TradeIdBubble } from "./TradeIdBubble";

const goToContract = jest.fn();
jest.mock("../../../../hooks/useTradeNavigation");
const useNavigateToOfferOrContractMock = jest
  .requireMock("../../../../hooks/useTradeNavigation")
  .useTradeNavigation.mockReturnValue(goToContract);

describe("TradeIdBubble", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<TradeIdBubble trade={contractSummary} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it("calls bump fees", () => {
    const { getByText } = render(<TradeIdBubble trade={contractSummary} />);
    fireEvent.press(getByText("PC‑7B‑1C8"));

    expect(useNavigateToOfferOrContractMock).toHaveBeenCalledWith(
      contractSummary,
    );
    expect(goToContract).toHaveBeenCalled();
  });
});
