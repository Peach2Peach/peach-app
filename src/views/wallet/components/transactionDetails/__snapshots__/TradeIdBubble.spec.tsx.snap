// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeIdBubble renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FEEDE5",
      "borderColor": "#FCCCB6",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 4,
      "height": 28,
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2B1911",
          "fontFamily": "Baloo2-Regular",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 24,
        },
        [
          {
            "color": "#2B1911",
          },
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 14,
            "letterSpacing": 0.5,
            "lineHeight": 24,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 15,
            "letterSpacing": 0.54,
            "lineHeight": 26,
            "textTransform": "uppercase",
          },
        ],
        false,
      ]
    }
  >
    PC‑7B‑1C8
  </Text>
  <test-file-stub
    fill="#F56522"
    style={
      [
        {
          "height": 16,
          "width": 16,
        },
        undefined,
      ]
    }
  />
</View>
`;
